cpu/samd5x/periph/can.c

```diff
@@ -187,15 +188,37 @@ static int _set_mode(Can *can, can_mode_t can_mode)
 
 static void _setup_clock(can_t *dev)
 {
-    if (dev->conf->can == CAN0) {
-        GCLK->PCHCTRL[CAN0_GCLK_ID].reg = GCLK_PCHCTRL_CHEN | GCLK_PCHCTRL_GEN(dev->conf->gclk_src);
+    int pchid = 0;
+    if (dev->conf->can == CAN0){
+        pchid = CAN0_GCLK_ID;
     }
-    else if (dev->conf->can == CAN1) {
-        GCLK->PCHCTRL[CAN1_GCLK_ID].reg = GCLK_PCHCTRL_CHEN | GCLK_PCHCTRL_GEN(dev->conf->gclk_src);
+    else if (dev->conf->can == CAN1){
+        pchid = CAN1_GCLK_ID;
     }
     else {
         DEBUG_PUTS("CAN channel not supported");
+        return;
     }
```

Comment:
```suggestion
    int pchid = CAN0_GCLK_ID;
    if (dev->conf->can != CAN0) {
        /* only CAN0 and CAN1 supported. When ported to MCUs with more
         * CAN controllers, this code needs to be adapted */
        assert(dev->conf->can == CAN1);
        pchid = CAN1_GCLK_ID;
    }
```

Reply:
i added an assert and comment but kept the structure